### PR TITLE
Add several spelling corrections

### PR DESCRIPTION
**Pull request by kianmeng**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -2708,6 +2708,7 @@ arre->are
 arreay->array
 arrengement->arrangement
 arrengements->arrangements
+arrity->arity
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:2711]: You learn something new...

Maybe add:
```suggestion
arrity->arity, parity,
```

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -15309,6 +15310,7 @@ funchtionns->functions
 funchtions->functions
 funcion->function
 funcions->functions
+funciotns->functions
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:15313]: Add the singular:
```suggestion
funciotn->function
funciotns->functions
```

**Review by peternewman — CHANGES_REQUESTED** [review]
A few minor tweaks.

**Comment by luzpaz** [comment]
Apologies, I though this was one of my personal PRs. I merged the changes @peternewman requested. :cringe:

**Review by peternewman — APPROVED** [review]
LGTM, but it would be nice to check that @kianmeng is happy with @luzpaz unexpected changes.

**Comment by luzpaz** [comment]
Yes, please weigh-in @kianmeng

**Comment by kianmeng** [comment]
@peternewman @luzpaz I'm good. :smile:

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**